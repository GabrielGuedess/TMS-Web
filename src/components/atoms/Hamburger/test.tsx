import { type ReactNode } from 'react';

import userEvent from '@testing-library/user-event';
import { render, screen, waitFor } from '@testing-library/react';

import { Hamburger } from '.';

jest.mock('next/link', () => ({
  __esModule: true,
  default: ({
    onClick,
    children,
  }: {
    children: ReactNode;
    onClick: () => void;
  }) => (
    <button type="button" onClick={onClick} data-testid="Mock Link">
      {children}
    </button>
  ),
}));

describe('<Hamburger />', () => {
  it('should render the hamburger correctly', () => {
    const { container } = render(<Hamburger />);

    expect(container.firstChild).toMatchSnapshot();
  });

  it('should be able change open/close the hamburger when clicked', async () => {
    render(<Hamburger />);

    const hamburger = screen.getByLabelText('Hamburger');

    await userEvent.click(hamburger);

    expect(hamburger.getAttribute('aria-expanded')).toBe('true');
  });

  it('should be able change close the hamburger when clicked "dashboard" option', async () => {
    render(<Hamburger />);

    const hamburger = screen.getByLabelText('Hamburger');

    await userEvent.click(hamburger);

    const item = screen.getAllByTestId('Mock Link')[0];

    expect(hamburger.getAttribute('aria-expanded')).toBe('true');

    await userEvent.click(item);

    await waitFor(() => {
      expect(hamburger.getAttribute('aria-expanded')).toBe('false');
    });
  });
});
