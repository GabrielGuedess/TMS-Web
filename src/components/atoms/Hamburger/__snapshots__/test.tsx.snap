// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Hamburger /> should render the hamburger correctly 1`] = `
<div
  class="z-50"
>
  <div
    aria-expanded="false"
    aria-label="Hamburger"
    class="hamburger-react"
    role="button"
    style="cursor: pointer; height: 48px; position: relative; transition: 0.4s cubic-bezier(0, 0, 0, 1); user-select: none; width: 48px; outline: none; transform: none;"
    tabindex="0"
  >
    <div
      style="background: currentColor; height: 3px; left: 8px; position: absolute; width: 32px; top: 13px; transition: 0.4s cubic-bezier(0, 0, 0, 1); transform: none;"
    />
    <div
      style="background: currentColor; height: 3px; left: 8px; position: absolute; width: 32px; top: 23px; transition: 0.4s cubic-bezier(0, 0, 0, 1); transform: none;"
    />
    <div
      style="background: currentColor; height: 3px; left: 8px; position: absolute; width: 32px; top: 33px; transition: 0.4s cubic-bezier(0, 0, 0, 1); transform: none;"
    />
  </div>
</div>
`;
